b2sdk/sync/sync.py

```diff
@@ -216,16 +134,149 @@ def sync_folders(
-        dest_folder.ensure_present()
+    if args.delete and (args.keepDays is not None):
+        raise CommandError('--delete and --keepDays are incompatible')
 
-    if source_folder.folder_type() == 'local' and not allow_empty_source:
-        source_folder.ensure_non_empty()
+    keep_days_or_delete = None
+    keep_days = None
+
+    if args.delete:
+        keep_days_or_delete = Synchronizer.DELETE_MODE
+    elif args.keepDays:
+        keep_days_or_delete = Synchronizer.KEEP_DAYS_MODE
+        keep_days = args.keepDays
+
+    synchronizer = Synchronizer(
+        no_progress,
+        max_workers,
+        policies_manager=policies_manager,
+        dry_run=False,
+        allow_empty_source=False,
+        newer_file_mode=newer_file_mode,
+        keep_days_or_delete=keep_days_or_delete,
+        compare_version_mode=AbstractFileSyncPolicy.COMPARE_VERSION_MODTIME,
+        compare_threshold=0,
+        keep_days=keep_days,
+    )
 
     # Make a reporter to report progress.
     with SyncReport(stdout, no_progress) as reporter:
+        try:
+            synchronizer.sync_folders(source_folder, dest_folder, now_millis, reporter)
+        except InvalidArgument as e:
+            raise CommandError('--%s %s', (e.field_name, e.message))
+
+
+class Synchronizer(object):
+    DELETE_MODE = 301
+    KEEP_DAYS_MODE = 302
+
+    KEEP_OR_DELETE_MODES = [DELETE_MODE, KEEP_DAYS_MODE]
+
+    def __init__(
+            self,
+            no_progress,
+            max_workers,
+            policies_manager=DEFAULT_SCAN_MANAGER,
+            dry_run=False,
+            allow_empty_source=False,
+            newer_file_mode=None,
+            keep_days_or_delete=None,
+            compare_version_mode=AbstractFileSyncPolicy.COMPARE_VERSION_MODTIME,
+            compare_threshold=0,
+            keep_days=None,
+    ):
+        """
+        Initialize synchronizer class and validate arguments
+
+        :param no_progress: if True, do not show progress
+        :type no_progress: bool
+        :param max_workers: max number of workers
+        :type max_workers: int
+        :param policies_manager: policies manager object
+        :param dry_run:
+        :type dry_run: bool
+        :param allow_empty_source: if True, do not check whether source folder is empty
+        :type allow_empty_source: bool
+        :param newer_file_mode: one of 101 (Skip) or 102 (Replace), can be None
+        :type newer_file_mode: int
+        :param keep_days_or_delete: one of 301 (Delete policy), 302 (keep for days policy), can be None
+        :type keep_days_or_delete: int
+        :param compare_version_mode: one of 201 (Modification time), 201 (Size), can be None, default is 201
+        :type compare_version_mode: int
+        :param compare_threshold: should be greater than 0, default is 0
+        :type compare_threshold: int
+        :param keep_days: if keep_days_or_delete is 302, then this should be greater than 0, default is None
```

Comment:
```suggestion
        :param int keep_days: if keep_days_or_delete is 302, then this should be greater than 0, default is None
```
is much more concise. I've already modified all docstrings in the project to follow this style, except for sync (to avoid conflict with your changes)

Reply:
@ppolewicz I think after this PR is finalized. I can create another PR to update all the doc strings to new style.